src/handler.lua

```diff
@@ -0,0 +1,161 @@
+local ValidateToken = require("kong.plugins.base_plugin"):extend()
+
+local http = require "resty.http"
+local cjson = require "cjson.safe"
+
+local find = string.find
+local format = string.format
+
+local TYPE_JSON = "application/json"
+
+local OPERATIONAL_OUTCOME_TEMPLATE =
+  '{ "resourceType": "OperationOutcome",\n' ..
+  '  "id": "exception",\n' ..
+  '  "text": {\n' ..
+  '      "status": "additional",\n' ..
+  '      "div": "<div xmlns=\\"http://www.w3.org/1999/xhtml\\"><p>%s</p></div>"\n' ..
+  '  },\n' ..
+  '  "issue": [\n' ..
+  '      {\n' ..
+  '          "severity": "error",\n' ..
+  '          "code": "exception",\n' ..
+  '          "details": {\n' ..
+  '              "text": "%s"\n' ..
+  '          }\n' ..
+  '      }\n' ..
+  '  ]\n' ..
+  '}'
+
+local INVALID_TOKEN = "invalid token."
+local BAD_VALIDATE_ENDPOINT = "Validate endpoint not found."
+local VALIDATE_ERROR = "Error validating token."
+local TOKEN_MISMATCH = "Token not allowed access to this patient."
+local SCOPE_MISMATCH = "Token not granted requested scope."
+
+function ValidateToken:new()
+  ValidateToken.super.new(self, "validate-token")
+
+end
+
+function ValidateToken:access(conf)
+  ValidateToken.super.access(self)
+
+  self.conf = conf
+
+  if (ngx.req.get_headers()["Authorization"] == nil) then
+      return self:send_response(401, INVALID_TOKEN)
+  end
+
+  local client = http.new()
+  client:set_timeout(self.conf.verification_timeout)
+
+  local verification_res, err = client:request_uri(self.conf.verification_url, {
+    method = "GET",
+    ssl_verify = false,
+    headers = {
+      Authorization = ngx.req.get_headers()["Authorization"],
+      apiKey = self.conf.api_key,
```

Comment:
When deployed to api.va.gov Kong gateway, the token will not be needed when invoking the validation endpoint because that token is enforced by Kong itself. So would be nice for it to have a default value in the schema.lua

Reply:
No, I mean just provide a default value in schema.lua, so that it would become optional in the configuration. We can also just configure the empty string, but in practice it feels like an optional param.

Reply:
My understanding was the `apiKey` was a required parameter for the `/validate` endpoint.  I didn't default the value since I didn't want it to be accidentally omitted.  Is that not the case?